### ISSUE TITLE
Open Source Licence for Tamarin Manual

### ISSUE THREAD

**Issue by mpdehnel** [tamarin-prover/manual#33]
**TL;DR: Could someone _pretty please_ add a formal licence to the manual?**  :-)(Preferably/hopefully/ideally something like GPLv3 as per the rest of Tamarin).

The [Wikipedia page for Tamarin](https://en.wikipedia.org/wiki/Tamarin_Prover) has sadly (erroneously) been marked for deletion due to alleged "copyright violation"; they are correct in pointing out that the wiki article text quotes a couple of paragraphs of the manual _verbatim_, as I thought it was well written and useful content. 

It is my belief that the manual was written and released under an open source licence (same as Tamarin itself, hence the repo being publicly available here on GitHub), and that this use of the text is not a copyright violation. If I can prove that the content is open source, this issue goes away. 

To resolve this and any future issues, would it be possible to include a formal `LICENCE` file (preferably licensing it as something like GPLv3, although obviously whatever you think most suitable!) within this public repo, so that I can demonstrate its open source status to the Wikipedia administrators? I really don't want my lovely article to be deleted!

Thanks, and hope you've all had a lovely Christmas / holiday period.

Martin

P.S. I assume the `LICENCE.highlight.js` is a licence for `highlight.js` or other such software used to create the manual, and not the the text of the manual as a whole!

**Comment by mpdehnel** [comment]
Update: the editor who flagged it has now formally requested proof of the open source status, so if it is the case that it's open source, I'd be really grateful for the licence being updated. If it's not, I'll re-write as much of the article as possible.

**Comment by rsasse** [comment]
Hi Martin,

Thanks for your efforts with this! The manual is under open source 
license, particularly BSD-3, visible at https://github.com/tamarin-prover

Also, there is this file which may help: 
https://github.com/tamarin-prover/manual/blob/master/LICENCE.highlight.js

Not sure if and what our reasoning was for picking this particular 
license and making it hard to find, so I need to hear from the others 
first before addressing this.

Cheers,
Ralf

**Comment by mpdehnel** [comment]
Was the BSD-3 licence a specific choice, or was it inferred by GitHub (correctly or otherwise) due to the inclusion of a BSD-Licensed bit of software in the repo, [`highlight.js`](https://highlightjs.org/)?

**Comment by rsasse** [comment]
Actually it might be incorrectly inferred, good point! Take a look at 
the PDF of the manual, particularly page 2 of: 
https://tamarin-prover.github.io/manual/tex/tamarin-manual.pdf

It says:

> This written work is licensed under a Creative Commons 
> Attribution-NonCommercial-ShareAlike 4.0 International License. You 
> may reproduce and edit this work with attribution for all 
> non-commercial purposes.

which is in line with what I remembered (i.e., a CC license). This 
should be the correct answer, but I am happy to hear from anyone else.

Cheers,
Ralf

**Comment by mpdehnel** [comment]
Perfect. Wikipedia may complain as this licence is strictly not compatible with theirs (I think you have to be CC BY-SA rather than CC BY-NC-SA) but this should be enough to get them off my back for now! Could this licence be added to the repo as a separate `LICENCE` file, and included clearly in the HTML version? Thank you!

**Comment by katrielalex** [comment]
NC is a weird licence for the manual, was that intentional? I'd have
thought commercial use was encouraged...
K


On Sat, 30 Dec 2017, at 5:14 PM, Martin Dehnel-Wild wrote:
> Perfect. Wikipedia may complain as this licence is strictly not
> compatible with theirs (I think you have to be CC BY-SA rather than CC
> BY-NC-SA) but this should be enough to get them off my back for now!
> Could this licence be added to the repo as a separate LICENCE file,
> and included clearly in the HTML version? Thank you!> — You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub[1], or mute the
> thread[2].> 


Links:

  1. https://github.com/tamarin-prover/manual/issues/33#issuecomment-354557060

**Comment by mpdehnel** [comment]
CC BY-NC-SA:
>This license lets others remix, tweak, and build upon your work non-commercially, and although their new works must also acknowledge you and be non-commercial, they don’t have to license their derivative works on the same terms.
I don't think this means people can't use the material if they're a commercial entity, just that they can't modify it and then use the result commercially?

**Comment by mpdehnel** [comment]
Ok, I'm pleased to note that a wikipedia administrator has removed the 'mark for deletion (copyright violation)' flag, and downgraded it to a 'copypaste warning', i.e. it has been noted that some of the content on this page has been copied and pasted from another source, and that this content _may_ not be compatible with wikipedia's licenses. 
>This article or section may have been copied and pasted from https://tamarin-prover.github.io/manual/book/001_introduction.html (DupDet · CopyVios), possibly in violation of Wikipedia's copyright policy. Please remedy this by editing this article to remove any non-free copyrighted content and attributing free content correctly, or flagging the content for deletion.

This gives us a decent stay of execution on this front. I will spend a bit of free time (when I have it) re-phrasing / writing sections so that it's no longer so close to the manual.

I would still really appreciate the licence being written into the HTML version and the source repo formally! Thank you :-)

**Comment by mpdehnel** [comment]
Ok, so to further muddy the waters: the [CC BY-NC-SA](https://creativecommons.org/licenses/by-nc-sa/4.0/) licence defines the non-commercial bit as follows:
>NonCommercial — You may not use the material for commercial purposes.

Does this mean that the Tamarin Manual may not be used (i.e. read / used / relied upon / quoted by) any entity wishing to use it for commercial gain? If I do some consultancy work with Tamarin, am I therefore not allowed to read the manual to support this work? I'm a bit confused by this. If this is the correct interpretation of this licence, it seems like an oversight. If not, some clarification would be useful!

**Comment by jdreier** [comment]
Tamarin itself is under a GPL license, which also permits commercial use. Only the manual is limited to non-commercial use to prevent e.g. somebody printing it and selling it as a book. However, in my understanding, a company employee can still read the manual and use Tamarin as part of his work (but I am not a lawyer).

**Comment by mpdehnel** [comment]
PR adding licence to HTML version now accepted.
